class DataSourceError(Exception):
    """Custom exception for data source errors."""

    pass

class BadHashError(Exception):
    """Custom exception for failed hash match."""

    pass
